### PR TITLE
feat!: added catalogId and displayPath support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Add tiles path field to catalog record

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -201,4 +201,7 @@ components:
           type: array
           items:
             $ref: ./Schema/link.yaml#/components/schemas/Link
+        displayPath:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:204]: why is it here instead of the mc-models?

File: src/DAL/repositories/recordRepository.ts [hunk 2]
```diff
@@ -22,14 +22,16 @@ export class RecordRepository extends Repository<RecordEntity> {
 
   public async createRecord(req: IRasterCatalogUpsertRequestBody): Promise<string> {
     const entity = this.recordConvertor.createModelToEntity(req);
+    entity.id = req.id;
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/repositories/recordRepository.ts:25]: the converter don't includes the id already?

File: src/common/dataModels/records.ts [hunk 3]
```diff
@@ -23,4 +23,4 @@ export interface IRecordExistsResponse {
   exists: boolean;
 }
 
-export interface IFindRecordResponse extends IRecordRequestParams, Partial<IRasterCatalogUpsertRequestBody> {}
+export interface IFindRecordResponse extends IRasterCatalogUpsertRequestBody {}
```
Comment (cl-shaharshavit-rnd) [inline src/common/dataModels/records.ts:26]: why?

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -51,7 +51,7 @@
     "@map-colonies/error-types": "^1.1.0",
     "@map-colonies/express-access-log-middleware": "^0.0.3",
     "@map-colonies/js-logger": "^0.0.3",
-    "@map-colonies/mc-model-types": "^12.1.0",
+    "@map-colonies/mc-model-types": "file:../mc-models/map-colonies-mc-model-types-13.1.3.tgz",
```
Comment (asafMasa) [inline package.json:54]: replcae when mc-models will be published

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (asafMasa) [reply package.json:54]: fix lint and tests

Comment (asafMasa) [reply package.json:54]: remove auto-generated ID creation

File: openapi3.yaml (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml]: fixed

File: package.json (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply package.json:54]: id default value removed from db schema

File: src/DAL/repositories/recordRepository.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/DAL/repositories/recordRepository.ts]: fixed

File: src/common/dataModels/records.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/dataModels/records.ts:26]: 'IRasterCatalogUpsertRequestBody' is now including "id" field so its unnecessary extends 'IRecordRequestParams'

Comment (cl-shaharshavit-rnd) [reply src/common/dataModels/records.ts:26]: use Partial<IRasterCatalogUpsertRequestBody>

File: src/DAL/convertors/recordModelConverter.ts [hunk 5]
```diff
@@ -56,6 +56,9 @@ export class RecordModelConvertor {
 
   private parseMetadata(entity: RecordEntity, metadata: Partial<LayerMetadata>): void {
     Object.assign(entity, metadata);
+    if (metadata.id != undefined) {
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/convertors/recordModelConverter.ts:59]: check if necessary and remove condition

File: src/DAL/migration/v2.2.0.sql [hunk 6]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE "records" ALTER identifier DROP DEFAULT;
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/migration/v2.2.0.sql:1]: this should be major as it is breaking change

File: src/DAL/repositories/recordRepository.ts [hunk 7]
```diff
@@ -22,14 +22,16 @@ export class RecordRepository extends Repository<RecordEntity> {
 
   public async createRecord(req: IRasterCatalogUpsertRequestBody): Promise<string> {
     const entity = this.recordConvertor.createModelToEntity(req);
+    console.log("ENTITY:", entity)
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/repositories/recordRepository.ts:25]: console

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]
add ! to pr name to make it major

File: src/DAL/repositories/recordRepository.ts (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/DAL/repositories/recordRepository.ts]: removed

File: src/DAL/migration/v2.2.0.sql (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/DAL/migration/v2.2.0.sql]: fixed

**Review by asafMasa — APPROVED** [review]

**Review by cl-shaharshavit-rnd — APPROVED** [review]

**Pull request merged**